File: app/src/main/java/com/gorvi/fastingtracker/FastingRecord.java
package com.gorvi.fastingtracker;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import java.time.LocalDateTime;

@Entity(tableName = "fasting_records")
public class FastingRecord {
    @PrimaryKey(autoGenerate = true)
    private long id;
    private LocalDateTime startTime;
    private LocalDateTime endTime;

    public FastingRecord(LocalDateTime startTime, LocalDateTime endTime) {
        this.startTime = startTime;
        this.endTime = endTime;
    }

    public long getId() {
        return id;
    }

    public LocalDateTime getStartTime() {
        return startTime;
    }

    public LocalDateTime getEndTime() {
        return endTime;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setStartTime(LocalDateTime startTime) {
        this.startTime = startTime;
    }

    public void setEndTime(LocalDateTime endTime) {
        this.endTime = endTime;
    }

}

File: app/src/main/java/com/gorvi/fastingtracker/FastingRecord.kt
package com.gorvi.fastingtracker

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.Duration
import java.time.LocalDateTime

@Entity(tableName = "fasting_records")
class FastingRecord(var startTime: LocalDateTime, var endTime: LocalDateTime?) {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    fun duration(): Duration =
        endTime?.let { Duration.between(startTime, it) } ?: Duration.ZERO


    override fun toString(): String {
        return "Start Time: ${this.startTime}\n" +
                "End Time: ${this.endTime}\n" +
                "Duration: ${this.duration().toString().substring(2)}"
    }
}